### PR TITLE
feat: service accounts (hooks)

### PR DESCRIPTION
**Pull request by nunogois**
https://linear.app/unleash/issue/2-539/hooks-for-service-accounts

Creates and adapts hooks for the Service Accounts feature, to be used by the FE.

### CODE REVIEW

File: frontend/src/hooks/api/actions/usePersonalAPITokensApi/usePersonalAPITokensApi.ts [hunk 1]
```diff
@@ -37,9 +37,26 @@ export const usePersonalAPITokensApi = () => {
...
+        const req = createRequest(`api/admin/user-admin/${userId}/pat`, {
+            method: 'POST',
+            body: JSON.stringify(payload),
+        });
+        try {
+            const response = await makeRequest(req.caller, req.id);
+            return await response.json();
+        } catch (e) {
+            throw e;
+        }
```
Comment (gastonfournier) [inline frontend/src/hooks/api/actions/usePersonalAPITokensApi/usePersonalAPITokensApi.ts:53]: Out of curiosity, wouldn't this be the same?
```suggestion
        const response = await makeRequest(req.caller, req.id);
        return await response.json();
```

File: frontend/src/hooks/api/actions/useServiceAccountsApi/useServiceAccountsApi.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import useAPI from '../useApi/useApi';
+
+export interface IServiceAccountPayload {
+    name: string;
+    username: string;
+    rootRole: number;
```
Comment (gastonfournier) [inline frontend/src/hooks/api/actions/useServiceAccountsApi/useServiceAccountsApi.ts:6]: Not sure what the `rootRole` is, just asking to learn about it.

**Review by gastonfournier — APPROVED** [review]
Just a small comment that won't make a difference, and a question for me to learn ;) Looks good!

File: frontend/src/hooks/api/actions/usePersonalAPITokensApi/usePersonalAPITokensApi.ts (hunk 1 above)
Comment (nunogois) [reply frontend/src/hooks/api/actions/usePersonalAPITokensApi/usePersonalAPITokensApi.ts:53]: You're right that it should be the same, but this is consistent with the other hooks we have. No idea what the original reasoning was behind it.

File: frontend/src/hooks/api/actions/useServiceAccountsApi/useServiceAccountsApi.ts (hunk 2 above)
Comment (nunogois) [reply frontend/src/hooks/api/actions/useServiceAccountsApi/useServiceAccountsApi.ts:6]: `rootRole` should map to one of our root roles in Unleash: 
 - Admin;
 - Editor;
 - Viewer;
 
 Just like regular users, service accounts also have a root role associated with them: 
image

**Pull request merged**